cmake/BLTMacros.cmake

```diff
@@ -1401,3 +1401,21 @@ macro(blt_export_tpl_targets)
         endif()
     endforeach()
 endmacro()
+
+macro(blt_convert_to_system_includes)
+    set(options)
+    set(singleValuedArgs TARGET)
+    set(multiValuedArgs)
+
+    ## parse the arguments to the macro
+    cmake_parse_arguments(arg
+         "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    if(NOT DEFINED arg_TARGET)
+       message(FATAL_ERROR "TARGET is a required parameter for the blt_convert_to_system_includes macro.")
+    endif()
+
+    get_target_property(_include_dirs ${arg_TARGET} INTERFACE_INCLUDE_DIRECTORIES)
+    set_property(TARGET ${arg_TARGET} APPEND PROPERTY INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${_include_dirs}")
```

Comment:
Should the macro remove the original includes before appending the system ones?

This might help users better debug their compilation lines.

Reply:
See my latest change and let me know if that's what you had in mind.

Reply:
Can you try this? I was under the impression the directories had to be duplicated in both places:

https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_SYSTEM_INCLUDE_DIRECTORIES.html

Reply:
Thanks @white238 -- Looks like the direct answer to my question is 
"No, we should not remove them"  
(thanks for digging into that).

On the other hand, combining [your link](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_SYSTEM_INCLUDE_DIRECTORIES.html) and [this page](https://cmake.org/cmake/help/latest/command/target_include_directories.html#command:target_include_directories) (linked to from your link might) seems to suggest that we might want to use `target_include_directories(SYSTEM)` in the proposed macro.

Having said that, since the current version works, that can be something we investigate in the future

Reply:
I've always set it manually... for example:

https://github.com/LLNL/serac/blob/10635cd995fa88ebbc58794aeab38c97c07e055b/cmake/thirdparty/SetupSeracThirdParty.cmake#L17-L20

Also I just confirmed that:

```
target_include_directories(foo SYSTEM PUBLIC /some/path)
blt_print_target_properties(TARGET foo)
```

does populate both `INTERFACE_INCLUDE_DIRECTORES` and `INTERFACE_SYSTEM_INCLUDE_DIRECTORIES`. 

This is because the SYSTEM property marks the path for `-isystem` while `INCLUDE_DIRECTORIES` or `INTERFACE_INCLUDE_DIRECTORIES` tells it to use the path for this target or inherit.

Reply:
I changed it to `target_include_directories(foo SYSTEM **INTERFACE** /some/path)`

Reply:
Another question, do I need to recursively iterate through the INTERFACE_LINK_LIBRARIES and do the same thing on each of them?

Reply:
No.. this level is fine.